extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/AuthenticationFailureEvent.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.security.spi.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+/**
+ * Security event fired when authentication failed.
+ */
+public class AuthenticationFailureEvent implements SecurityEvent {
+
+    public static final String AUTHENTICATION_FAILURE_KEY = AuthenticationFailureEvent.class.getName() + ".FAILURE";
+    private final Map<String, Object> eventProperties;
```

Comment:
I wonder if it would make sense to create an `AbstractSecurityEvent` with `eventProperties` and `securityIdentity` fields?

Reply:
I had it, I changed it as I thought user will use `SecurityEvent` which as interface looks better in my eyes then `AbstractSecurityEvent`. But considering how much trouble I had to convince Sergey to keep this interface (as he expects users will prefer implementations), I can change it to abstract class. It will help to drop some code repetition.

Reply:
Nono, the `SecurityEvent` interface is perfect. I would personally keep it. I was more thinking about reusing the logic in the impl classes...

Reply:
Yeah, this way I like it, will do it.

Reply:
@michalvavrik @mkouba Thanks, indeed, interface should stay

Reply:
It was for better, done.